aws/resource_aws_dynamodb_table.go

```diff
@@ -619,6 +623,30 @@ func resourceAwsDynamoDbTableDelete(d *schema.ResourceData, meta interface{}) er
 
 	log.Printf("[DEBUG] DynamoDB delete table: %s", d.Id())
 
+	if backupName, ok := d.GetOk("final_backup_identifier"); ok {
```

Comment:
**NOTE:** One challenge we have with reading `ResourceData` with `Get*()` during a `Delete` function is that the state for that attribute _must_ have be applied beforehand with the expected value before the deletion is occurring. The attribute value cannot be added/updated at the same time as the deletion is occurring. This is a global problem we have with the Terraform Provider SDK at the moment, but it does make usage like this unexpected for operators, especially if someone is absolutely expecting a backup of their table to occur before deletion and Terraform then does not do that. 🙁